### PR TITLE
Feature Command Builders

### PR DESCRIPTION
**Pull request by MicheleRezk**
Adding command builders

### CODE REVIEW

File: commercetools.Sdk/commercetools.Sdk.HttpApi/CommandBuilders/CommandBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,825 @@
...
+            this UpdateCommandBuilder<T> builder,
+            Func<UpdateAction<T>> action)
+            where T : Resource<T>
+        {
+            var addedAction = action.Invoke();
+            builder.Actions.Add(addedAction);
+            return builder;
+        }
+
+        // public static UpdateCommandBuilder<UpdateCommand<Customer>, Customer> AddAddress(this UpdateCommandBuilder<UpdateCommand<Customer>, Customer> builder, Func<AddAddressUpdateAction, UpdateAction<Customer>> action)
```
Comment (jenschude) [inline commercetools.Sdk/commercetools.Sdk.HttpApi/CommandBuilders/CommandBuilderExtensions.cs]: please remove commented code blocks

File: commercetools.Sdk/commercetools.Sdk.HttpApi/CommandBuilders/CommandBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,825 @@
...
+        public static CommandBuilder<ReplicateCommand<Cart>, Cart> Replicate(
+            this DomainCommandBuilder<Cart> builder,
+            IReplicaDraft<Cart> replica)
+        {
+            return new CommandBuilder<ReplicateCommand<Cart>, Cart>(
+                builder.Client,
+                () => new ReplicateCartCommand(replica) as ReplicateCommand<Cart>);
+        }
+
+        /*
```
Comment (jenschude) [inline commercetools.Sdk/commercetools.Sdk.HttpApi/CommandBuilders/CommandBuilderExtensions.cs]: remove please

File: commercetools.Sdk/IntegrationTests/commercetools.Sdk.IntegrationTests/CommandBuilders/CommandBuildersIntegrationTests.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+using commercetools.Sdk.Client;
+using commercetools.Sdk.HttpApi;
+using commercetools.Sdk.HttpApi.CommandBuilders;
+using Xunit;
+using static commercetools.Sdk.IntegrationTests.Categories.CategoriesFixture;
+
+namespace commercetools.Sdk.IntegrationTests.CommandBuilders
+{
+    [Collection("Integration Tests")]
+    public class CommandBuildersIntegrationTests
```
[the remark this answers is not in the document]
Comment (MicheleRezk) [reply commercetools.Sdk/IntegrationTests/commercetools.Sdk.IntegrationTests/CommandBuilders/CommandBuildersIntegrationTests.cs]: it's deleted now

File: commercetools.Sdk/IntegrationTests/commercetools.Sdk.IntegrationTests/Customers/CustomersIntegrationTestsUsingCommandBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+using static commercetools.Sdk.IntegrationTests.CustomerGroups.CustomerGroupsFixture;
+using static commercetools.Sdk.IntegrationTests.Types.TypesFixture;
+using static commercetools.Sdk.IntegrationTests.Projects.ProjectFixture;
+using static commercetools.Sdk.IntegrationTests.Stores.StoresFixture;
+using static commercetools.Sdk.IntegrationTests.GenericFixture;
+
+namespace commercetools.Sdk.IntegrationTests.Customers
+{
+    [Collection("Integration Tests")]
+    public class CustomersIntegrationTestsUsingCommandBuilder
```
[the remark this answers is not in the document]
Comment (MicheleRezk) [reply commercetools.Sdk/IntegrationTests/commercetools.Sdk.IntegrationTests/Customers/CustomersIntegrationTestsUsingCommandBuilder.cs]: Done :thumbsup:

File: commercetools.Sdk/commercetools.Sdk.HttpApi/CommandBuilders/CommandBuilderExtensions.cs (hunk 1 above)
Comment (MicheleRezk) [reply commercetools.Sdk/commercetools.Sdk.HttpApi/CommandBuilders/CommandBuilderExtensions.cs]: :thumbsup:

File: commercetools.Sdk/commercetools.Sdk.HttpApi/CommandBuilders/CommandBuilderExtensions.cs (hunk 2 above)
Comment (MicheleRezk) [reply commercetools.Sdk/commercetools.Sdk.HttpApi/CommandBuilders/CommandBuilderExtensions.cs]: :thumbsup:

**Comment by MicheleRezk** [comment]
@jenschude it seems that the json serializer in system.text.json don't ignoring JsonValue in Attribute Model even if we're using ShouldSerializeJsonValue (which Ignored by Newtonsoft), I had to create custom converter for Attribute to Ignore it

**Pull request merged**